src/main/java/slimeknights/tconstruct/common/data/TConstructBlockLootTables.java

```diff
@@ -133,137 +123,103 @@ private void addTools() {
-    this.registerLootTable(WorldBlocks.purple_green_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.green_slime_dirt));
-    this.registerLootTable(WorldBlocks.orange_green_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.green_slime_dirt));
-
-    this.registerLootTable(WorldBlocks.blue_blue_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.blue_slime_dirt));
-    this.registerLootTable(WorldBlocks.purple_blue_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.blue_slime_dirt));
-    this.registerLootTable(WorldBlocks.orange_blue_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.blue_slime_dirt));
+    this.registerDropSelfLootTable(WorldBlocks.cobalt_ore.get());
+    this.registerDropSelfLootTable(WorldBlocks.ardite_ore.get());
 
-    this.registerLootTable(WorldBlocks.blue_purple_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.purple_slime_dirt));
-    this.registerLootTable(WorldBlocks.purple_purple_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.purple_slime_dirt));
-    this.registerLootTable(WorldBlocks.orange_purple_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.purple_slime_dirt));
-
-    this.registerLootTable(WorldBlocks.blue_magma_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.magma_slime_dirt));
-    this.registerLootTable(WorldBlocks.purple_magma_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.magma_slime_dirt));
-    this.registerLootTable(WorldBlocks.orange_magma_slime_grass, (block) -> droppingWithSilkTouch(block, WorldBlocks.magma_slime_dirt));
-
-    this.registerLootTable(WorldBlocks.blue_slime_leaves, (block) -> randomDropBlueOrGreenSlimeBall(block, WorldBlocks.blue_slime_sapling, DEFAULT_SAPLING_DROP_RATES));
-
-    this.registerLootTable(WorldBlocks.purple_slime_leaves, (block) -> randomDropPurpleSlimeBall(block, WorldBlocks.purple_slime_sapling, DEFAULT_SAPLING_DROP_RATES));
-
-    this.registerLootTable(WorldBlocks.orange_slime_leaves, (block) -> dropSapling(block, WorldBlocks.orange_slime_sapling, DEFAULT_SAPLING_DROP_RATES));
-
-    this.registerLootTable(WorldBlocks.blue_slime_fern, BlockLootTables::onlyWithShears);
-    this.registerLootTable(WorldBlocks.purple_slime_fern, BlockLootTables::onlyWithShears);
-    this.registerLootTable(WorldBlocks.orange_slime_fern, BlockLootTables::onlyWithShears);
-
-    this.registerLootTable(WorldBlocks.blue_slime_tall_grass, BlockLootTables::onlyWithShears);
-    this.registerLootTable(WorldBlocks.purple_slime_tall_grass, BlockLootTables::onlyWithShears);
-    this.registerLootTable(WorldBlocks.orange_slime_tall_grass, BlockLootTables::onlyWithShears);
+    for (SlimeBlock.SlimeType slime : SlimeBlock.SlimeType.values()) {
+      this.registerDropSelfLootTable(WorldBlocks.slime.get(slime));
```

Comment:
Skip vanilla green in this loop. Might be easier to do a separate loop, one over TINKERS slimes and one over all.